test/integration_test/basic_test.go

```diff
@@ -410,3 +418,64 @@ func BasicStorkRegression(tc *types.TestCase) func(*testing.T) {
 		ci_utils.UninstallAndValidateStorageCluster(cluster, t)
 	}
 }
+
+// BasicAutopilotRegression test includes the following steps:
+// 1. Deploy PX and validate Stork components and images
+// 2. Validate Autopilot is disabled by default
+// 3. Enabled Autopilot and validate Autopilot components and images
+// 4. Delete "autopilot" pod and validate it gets re-deployed
+// 5. Disable Autopilot and validate Autopilot components got successfully removed
+// 6. Delete StorageCluster and validate it got successfully removed
+func BasicAutopilotRegression(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate Autopilot block is nil
+		require.Nil(t, cluster.Spec.Autopilot, "failed to validate Autopilot block, it should be nil by default, but it seems there is something set in there %+v", cluster.Spec.Autopilot)
+
+		logrus.Info("Enable Autopilot and validate StorageCluster")
+		updateParamFunc := func(cluster *corev1.StorageCluster) *corev1.StorageCluster {
+			// At this point this object should be <nil>
+			if cluster.Spec.Autopilot == nil {
+				cluster.Spec.Autopilot = &corev1.AutopilotSpec{}
+			}
```

Comment:
nit: As we know the object is going to be nil, we can simply set the Enabled flag - 
```
cluster.Spec.Autopilot = &corev1.AutopilotSpec{
  Enabled: true
}
```

Reply:
Changed this to suggested